test/README.md

```diff
@@ -0,0 +1,56 @@
+# HWI Tests
+
+## Running the tests
+
+This folder contains test cases for HWI. To run these tests, `hwilib` will need to be installed to your python system. You can install it by doing `python3 setup.py install` in the root directory.
+
+- `test_bech32.py` tests the bech32 serialization.
+This is taken directly from the [python reference implementation](https://github.com/sipa/bech32/blob/master/ref/python/tests.py).
+- `test_psbt.py` tests the psbt serialization.
+It implements all of the [BIP 174 serialization test vectors](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#Test_Vectors).
+- `test_trezor.py` tests the command line interface and the Trezor implementation.
+It uses the [Trezor One firmware emulator](https://github.com/trezor/trezor-mcu/#building-for-development).
+It also tests usage with `bitcoind`, so the [patched Bitcoin Core](../docs/bitcoin-core-usage.md#bitcoin-core) is required.
+- `test_coldcard.py` tests the command line interface and Coldcard implementation.
+It uses the [Coldcard simulator](https://github.com/Coldcard/firmware/tree/master/unix#coldcard-desktop-simulator).
+It also tests usage with `bitcoind`, so the [patched Bitcoin Core](../docs/bitcoin-core-usage.md#bitcoin-core) is required.
+
+`setup_environment.sh` will build the Trezor emulator, the Coldcard simulator, and the patched `bitcoind`.
+if run in the `test/` directory, these will be built in `test/trezor-mcu`, `test/firmware`, and `test/bitcoind` respectively.
+
+`run_tests.py` runs the tests. If run from the `test/` directory, it will be able to find the Trezor emulator, Coldcard simulator, and bitcoind.
+Otherwise the paths to those will need to be specified on the command line.
+test_trezor.py` and `test_coldcard.py` can be disabled.
+
+## Trezor emulator dependencies
+
+In order to build the Trezor emulator, the following packages will need to be installed:
+
+```
+build-essential curl git python3 python3-pip libsdl2-dev libsdl2-image-dev gcc-arm-none-eabi libnewlib-arm-none-eabi
+```
+
+The following python packages will need to be installed
+
+```
+pipenv
+```
+
```

Comment:
You can build it without docker, unless you want to use `./build.sh`. AFAIK, `build.sh` does not build the emulator in headless mode, so there's a GUI and such. For automated testing on Travis, we need headless mode so we build it through the `pipenv` commands rather than `build.sh`.